ergotree-ir/src/mir/coll_append.rs

```diff
@@ -0,0 +1,124 @@
+use crate::serialization::op_code::OpCode;
+use crate::serialization::sigma_byte_reader::SigmaByteRead;
+use crate::serialization::sigma_byte_writer::SigmaByteWrite;
+use crate::serialization::SerializationError;
+use crate::serialization::SigmaSerializable;
+// use crate::types::stuple::STuple;
+use crate::types::stype::SType;
+
+
+use super::expr::Expr;
+use super::expr::InvalidArgumentError;
+use crate::has_opcode::HasStaticOpCode;
+
+/// Takes two collections as input and concatenates both into the Append Colelction
+#[derive(PartialEq, Eq, Debug, Clone)]
+pub struct Append {
+    /// Collection - First Parameter; first half of the combined collection
+    pub input: Box<Expr>,
+    /// Collection - Second Parameter; later half of the combined collection
+    pub output: Box<Expr>,
+}
+
+impl Append {
+    /// Create new object, returns an error if any of the requirements failed
+    pub fn new(input: Expr, output: Expr) -> Result<Self, InvalidArgumentError> {
+        let first_elem_type : SType = input.post_eval_tpe();
```

Comment:
`input_coll_type`?

Reply:
I find `first_elem_type` name confusing since IR node's arguments are collections. So I read it as we store the element type(as `T` in `Coll[T]`) of the `input` collection. I suggest to inline them and don't bother to name them at all.